lib/impress.client.js

```diff
@@ -733,26 +740,25 @@ Client.prototype.detectRealPath = function(
   let relPath = application.relative(client.realPathDir);
   const folderCache = application.cache.folders.get(relPath);
 
-  if (
+  const detected = (
     folderCache && folderCache !== impress.DIR_NOT_EXISTS ||
     client.realPath === '/'
-  ) {
-    callback();
-  } else {
-    api.fs.exists(client.realPathDir, (exists) => {
-      if (exists) {
-        application.cache.folders.set(relPath, impress.DIR_EXISTS);
-        callback();
-      } else {
-        application.cache.folders.set(relPath, impress.DIR_NOT_EXISTS);
-        client.realPath = api.common.dirname(client.realPath);
-        client.realPathDir = application.dir + '/www' + client.realPath;
-        relPath = application.relative(client.realPathDir);
-        client.detectRealPath(callback);
-      }
-      application.cache.watch(api.common.stripTrailingSlash(relPath));
-    });
-  }
+  );
+  if (detected) return callback();
+
+  api.fs.exists(client.realPathDir, (exists) => {
```

Comment:
`fs.exists` is considered deprecated for a long time now, it is better to use `fs.access` in this case.